commands/github-projects.js

```diff
@@ -0,0 +1,214 @@
+            const { restEndpointMethods } = require("@octokit/plugin-rest-endpoint-methods");
+            const MyOctokit = Octokit.plugin(restEndpointMethods);
+            let octokit = new MyOctokit({ auth: githubToken });
+
+            octokit.rest.projects.createForRepo({
+                owner: args[1],
+                repo: args[2],
+                name: projectTitle,
+              })
+                .then((result) => {
+                    return message.reply(
+                        "your new project '" +
+                            projectTitle +
+                            "' has been created in " +
+                            args[1] +
+                            "'s repo, " +
+                            args[2] +
+                            "."
+                    );
+                })
+                .catch((error) => {
+                    console.error(error);
+                    return message.reply(
+                        "creating your project was unsuccessful. Please ensure you have set your GitHub token using -github and entered the information in the correct order of organisation name, repo name and project title and then try again."
+                    );
+                });
+        }
+
+        // Create Column function
+        async function createColumn (columnName, args) {
+            // GitHub Variables
+            let githubToken = readToken();
+
+            // Intialise GitHub API
+            const { Octokit } = require("@octokit/core");
+            const { restEndpointMethods } = require("@octokit/plugin-rest-endpoint-methods");
+            const MyOctokit = Octokit.plugin(restEndpointMethods);
+            let octokit = new MyOctokit({ auth: githubToken });
+            octokit.rest.projects.createColumn({
+                project_id: args[1],
+                name: columnName,
+              })
+                .then((result) => {
+                    return message.reply("Your new column, " + columnName + " has been successfully created in project " + "#" + args[1] + "."
+                    );
+                })
+                .catch((error) => {
+                    console.error(error);
+                    return message.reply(
+                        "creating a column was unsuccessful. Please ensure you have set your GitHub token using -github and provided the correct project ID and a unique column name and then try again."
+                    );
+                });
+        }
+
+        // Get Project function
+        async function getProject (projectID, args) {
+            // GitHub Variables
+            let githubToken = readToken();
+
+            // Intialise GitHub API
+            const { Octokit } = require("@octokit/core");
+            const { restEndpointMethods } = require("@octokit/plugin-rest-endpoint-methods");
+            const MyOctokit = Octokit.plugin(restEndpointMethods);
+            let octokit = new MyOctokit({ auth: githubToken });
```

Comment:
This code block is repeated several times throughout this (and other) files. Can you find a way to avoid duplicating it?

Take a read of this: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Reply:
Have refactored to avoid repeating the `readToken()` function but the Octokit requires are more complicated because there seems to be a syntax error when using the destructuring syntax `{}` with `module.exports`.